stores/useGovernanceAssetsStore.tsx

```diff
@@ -154,50 +178,73 @@ const getTokenAccountObj = async (
   governance: ProgramAccount<Governance>,
   tokenAccount: TokenProgramAccount<AccountInfo>,
   mintAccounts: TokenProgramAccount<MintInfo>[]
-) => {
+): Promise<AccountTypeNFT | AccountTypeToken | null> => {
   const isNftAccount =
     tokenAccount.account.mint.toBase58() === DEFAULT_NFT_TREASURY_MINT
-  const mint = mintAccounts.find(
-    (x) => x.publicKey.toBase58() === tokenAccount.account.mint.toBase58()
-  )
+
+  const mint = mintAccounts.find((x) =>
+    x.publicKey.equals(tokenAccount.account.mint)
+  )!
+
   if (isNftAccount) {
-    return new AccountTypeNFT(tokenAccount, mint!, governance)
+    return new AccountTypeNFT(tokenAccount, mint, governance)
   }
 
   if (
-    mint?.account.supply &&
-    mint?.account.supply.cmpn(1) !== 0 &&
+    mint.account.supply &&
+    mint.account.supply.cmpn(1) !== 0 &&
     mint.publicKey.toBase58() !== DEFAULT_NATIVE_SOL_MINT
   ) {
     return new AccountTypeToken(tokenAccount, mint!, governance)
   }
+
+  return null
 }
 
 const getSolAccountsObj = async (
   connection: ConnectionContext,
   accounts: AssetAccount[],
   solAccountsInfo: SolAccInfo[],
-  mintAccounts: TokenProgramAccount<MintInfo>[],
+  mintAccounts: TokenProgramAccount<MintAccount>[],
   governances: ProgramAccount<Governance>[]
-) => {
-  const solAccs: AccountTypeSol[] = []
-  for (const i of solAccountsInfo) {
-    const mint = mintAccounts.find((x) => x.publicKey.toBase58() === WSOL_MINT)
-    const governance = governances.find(
-      (x) => x.pubkey.toBase58() === i.governancePk.toBase58()
-    )
+): Promise<AssetAccount[]> => {
+  const solAccounts: AccountTypeSol[] = []
+
+  const wsolMintAccount = mintAccounts.find(
+    (x) => x.publicKey.toBase58() === WSOL_MINT
+  )! // WSOL should be here
+
+  for (const solAccountInfo of solAccountsInfo) {
+    const governance = governances.find((x) =>
+      x.pubkey.equals(solAccountInfo.governancePk)
+    )! // Governance should be here
+
     const account = await getSolAccountObj(
-      governance!,
+      governance,
       connection,
-      mint!,
+      wsolMintAccount,
       accounts,
-      i
+      solAccountInfo
     )
+
     if (account) {
-      solAccs.push(account)
+      solAccounts.push(account)
     }
   }
-  return solAccs as AssetAccount[]
+
+  return solAccounts
+}
+
+// Return array without duplicates
+function uniquePublicKey(array: PublicKey[]): PublicKey[] {
+  return Array.from(
+    array.reduce((mintsPks, publicKey) => {
+      // Transform to string for Set to be able to identify duplicates
+      mintsPks.add(publicKey.toBase58())
+
+      return mintsPks
+    }, new Set<string>())
+  ).map((address) => new PublicKey(address))
 }
 
 const getTokenAssetAccounts = async (
```

Comment:
That'd be great to stick one style of function declaration, either function xx (){} or const xx = () => {}